server/src/test/java/io/spine/server/ServerEnvironmentTest.java

```diff
@@ -198,14 +218,65 @@ void resetEnvironment() {
         void getSet() {
             StorageFactory factory = new MemoizingStorageFactory();
 
-            serverEnvironment.configureStorageForTests(factory);
+            serverEnvironment.use(factory, Tests.class);
             assertThat(((SystemAwareStorageFactory) serverEnvironment.storageFactory()).delegate())
                     .isEqualTo(factory);
         }
+
+        @Test
+        @DisplayName("using a deprecated method")
+        @SuppressWarnings("deprecation")
+        void deprecatedMethod() {
+            MemoizingStorageFactory factory = new MemoizingStorageFactory();
+
+            serverEnvironment.configureStorageForTests(factory);
+            StorageFactory configuredFactory = serverEnvironment.storageFactory();
+            assertThat(configuredFactory).isInstanceOf(SystemAwareStorageFactory.class);
+            assertThat(((SystemAwareStorageFactory) configuredFactory).delegate())
+                    .isSameInstanceAs(factory);
+        }
+    }
+
+    @Nested
+    @DisplayName("configure `StorageFactory` for a custom environment")
+    class LocalStorageFactoryConfig {
+
+        @BeforeEach
+        void reset() {
+            serverEnvironment.reset();
+            Environment.instance()
+                       .register(new Local());
+            Environment.instance()
+                       .setTo(new Local());
```

Comment:
See how weird this is? You have several pieces of the same inconvenient code in this test. How much easier that would be if you could treat `setTo` as registration as well?